core-rust/core-api-server/src/core_api/paging.rs

```diff
@@ -0,0 +1,277 @@
+use radix_engine::types::*;
+use std::ops::Add;
+use std::time::{Duration, Instant};
+
+/// A sequence of deterministically-ordered elements, which knows how to efficiently start iteration
+/// from an arbitrarily requested item.
+pub trait RandomAccessIterable<T, I: Iterator<Item = T>> {
+    /// Turns itself into an iterator which:
+    /// - either starts at the beginning of the sequence (if `from.is_none()`),
+    /// - or starts at the given `from` element (if it is actually exists in the sequence),
+    /// - or at the first actually-existing element after the given `from` (if the from was to be
```

Comment:
basically in that last point I was trying to cover the following case:

The sequence is `[m, u, s, e]`.
I come in and say: "hi, please give me iterator starting from `o`".

I have no idea that `o` is not in the sequence. Maybe I thought it is there, maybe it was there in the past - doesn't matter.
Still, the question is: where should the iterator start?

The sequence has some order of elements (knows it internally), but I (from the outside) can make no assumptions about the order of elements in the sequence`*`.

And the contract defines the following response: "imagine that `o` existed in the sequence, and start iteration from the next element after that imaginary `o`". The `o` kinda "has its place", it is just not present there. So the starting `o` point is "exclusive", but only because `o` isn't there, so it can't be "inclusive".

OFC if I asked for "start from `s`", I would get `[s, e]` back (inclusive).

`*` This actually matches our impl e.g. in JMT (which uses a prefix of 20 hashed bytes for JMT's reasons).